src/signal.cr

```diff
@@ -123,6 +124,10 @@ enum Signal : Int32
   end
 
   @@setup_default_handlers = Atomic::Flag.new
+  @@setup_sigfault_handler = Atomic::Flag.new
+  @@sigfault_handler = LibC::SigactionHandlerT.new { |sig, info, data|
+    __crystal_sigfault_handler(sig, info.value.si_addr)
```

Comment:
Do we need to keep `fun __crystal_sigfault_handler` around? It seems it was only necessary to export as a function for libcrystal to call it. Now we don't need that any more, the implementation can be moved completely into a Crystal method. Or directly to this proc here.

This could also be a follow-up refactoring.

Reply:
Yeah directly in the proc would make sense to me. I'll do some looking to see if anything else is depending on `__crystal_sigfault_handler` (either within crystal or naughty libraries 😄 )

Reply:
I don't think the proc needs to be assigned to a class var either, it can just be initialized directly inside the setup method.

Reply:
My understand of `proc` callbacks passed to C is that we still need a reference to it on the class (or somewhere else in Crystal-land), otherwise it might be subject to GC?

Reply:
Oh, yeah right.